packages/devtools_app/lib/src/timeline/flutter/flutter_frames_chart.dart

```diff
@@ -2,20 +2,669 @@
+  final List<BarEntry> _frameDurations = <BarEntry>[];
+
+  // Trace #1 UI Duration.
+  BarDataSet frameDurationsSet;
+
+  @override
+  void initState() {
+    _initController();
+
+    // TODO(terry): Pass true to simulate live feed.
+    _initData(true);
+
+    super.initState();
+  }
+
+  final lightTypeFace = TypeFace(
+    fontFamily: 'OpenSans',
+    fontWeight: FontWeight.w100,
+  );
+
+  final boldTypeFace = TypeFace(
+    fontFamily: 'OpenSans',
+    fontWeight: FontWeight.w800,
+  );
+
+  final double groupSpace = 0.04;
+  final double barSpace = 0;
+
+  void _initController() {
+    final desc = Description()..enabled = false;
+    _chartController = BarChartController(
+      axisLeftSettingFunction: (axisLeft, controller) {
+        axisLeft
+          ..typeface = lightTypeFace
+          ..drawGridLines = false
+          ..setStartAtZero(true)
+          ..addLimitLine(LimitLine(60, '60 FPS')
+            // TODO(terry): LEFT_TOP is clipped need to fix in MPFlutterChart.
+            ..labelPosition = LimitLabelPosition.RIGHT_TOP
+            ..textSize = 10
+            ..typeface = boldTypeFace
+            ..enableDashedLine(5, 5, 0)
+            ..lineColor = ColorUtils.HOLO_RED_LIGHT);
+      },
+      axisRightSettingFunction: (axisRight, controller) {
+        axisRight.enabled = false;
+      },
+      xAxisSettingFunction: (XAxis xAxis, controller) {
+        xAxis.enabled = true;
+        xAxis.drawLabels = true;
+        xAxis.setLabelCount1(3);
+        xAxis.position = XAxisPosition.BOTTOM;
+      },
+      legendSettingFunction: (legend, controller) {
+        legend.enabled = false;
+      },
+      drawGridBackground: false,
+//      dragXEnabled: true,
+//      dragYEnabled: true,
+//      scaleXEnabled: true,
+//      scaleYEnabled: true,
+//      pinchZoomEnabled: false,
+//      maxVisibleCount: 60,
+      drawBarShadow: false,
+      description: desc,
+      marker: CustomDataValues(onSelected: frameSelected),
+    );
+
+    // Compute padding around chart.
+    _chartController.setViewPortOffsets(50, 10, 10, 30);
+  }
+
+  @override
+  void dispose() {
+    _timer?.cancel();
+    super.dispose();
+  }
+
+  void frameSelected(int frameIndex) {
+    print('Bar Charted item selected frame index = $frameIndex');
+  }
+
+  // Light Blue 50 - 200
+  static const mainUiColorLight = Color.fromARGB(0xff, 0x81, 0xD4, 0xFA);
+
+  // Light Blue 50 - 700
+  static const mainGpuColorLight = Color.fromARGB(0xFF, 0x02, 0x88, 0xD1);
+
+  void _initData([bool simulateFeed = false]) {
+    // Prime the dataset with the first entry.  A missing entry will cause a failure
+    // with a NAN.
+    _frameDurations.add(createBarEntry(0));
+
+    if (!simulateFeed) {
+      _loadAllData(3);
+    } else if (_timer == null) {
+      _startFeed(3);
+    } else {
+      return;
+    }
+
+    // Create heap used dataset.
+    frameDurationsSet = BarDataSet(_frameDurations, 'UI')
```

Comment:
Why is the name here 'UI' when it includes UI and GPU frames?

Reply:
Oops should be durations (which its stacked).